yt_dlp/extractor/arte.py

```diff
@@ -5,195 +5,240 @@
     compat_str,
 )
 from ..utils import (
+    GeoRestrictedError,
     ExtractorError,
     int_or_none,
+    parse_iso8601,
     parse_qs,
-    qualities,
     strip_or_none,
     try_get,
-    unified_strdate,
     url_or_none,
 )
 
 
 class ArteTVBaseIE(InfoExtractor):
     _ARTE_LANGUAGES = 'fr|de|en|es|it|pl'
-    _API_BASE = 'https://api.arte.tv/api/player/v1'
+    _API_BASE = 'https://api.arte.tv/api/player/v2'
 
 
 class ArteTVIE(ArteTVBaseIE):
     _VALID_URL = r'''(?x)
-                    https?://
+                    (?:https?://
                         (?:
                             (?:www\.)?arte\.tv/(?P<lang>%(langs)s)/videos|
                             api\.arte\.tv/api/player/v\d+/config/(?P<lang_2>%(langs)s)
                         )
-                        /(?P<id>\d{6}-\d{3}-[AF])
+                    |arte://program)
+                        /(?P<id>\d{6}-\d{3}-[AF]|LIVE)
                     ''' % {'langs': ArteTVBaseIE._ARTE_LANGUAGES}
     _TESTS = [{
         'url': 'https://www.arte.tv/en/videos/088501-000-A/mexico-stealing-petrol-to-survive/',
         'info_dict': {
             'id': '088501-000-A',
-            'ext': 'mp4',
             'title': 'Mexico: Stealing Petrol to Survive',
+            'alt_title': 'ARTE Reportage',
+            'description': 'md5:35ec9baaa8ad0b2456447c7972ba3ca0',
+            'duration': 1428,
+            'thumbnail': 'https://api-cdn.arte.tv/api/mami/v1/program/en/088501-000-A/940x530?ts=1626083168',
             'upload_date': '20190628',
+            'timestamp': 1561759200,
+            'ext': 'mp4',
         },
     }, {
         'url': 'https://www.arte.tv/pl/videos/100103-000-A/usa-dyskryminacja-na-porodowce/',
-        'only_matching': True,
+        'info_dict': {
+            'id': '100103-000-A',
+            'title': 'USA: Dyskryminacja na porodówce',
+            'description': 'md5:242017b7cce59ffae340a54baefcafb1',
+            'alt_title': 'ARTE Reportage',
```

Comment:
this test fails

Reply:
Post verbose log.

Reply:
The title and alt-title are swapped. Should the field extracted be reversed? Or is it just the test defenition that is wrong?
```
❯ yt-dlp https://www.arte.tv/pl/videos/100103-000-A/usa-dyskryminacja-na-porodowce -O title= -O alt_title=
title = 'ARTE Reportage'
alt_title = 'USA: Dyskryminacja na porodówce'
```

Reply:
ARTE Reportage seems to be the title of the program and 'USA: Dyskryminacja na porodówce' the title of this specific program episode.

Reply:
Indeed. It seems I have mistakenly swapped them. Excerpt from the (reformatted) API endpoint response:

```json
{
  "data": {
    "id": "100103-000-A_pl",
    "type": "ConfigPlayer",
    "attributes": {
      "provider": "arte",
      "metadata": {
        "providerId": "100103-000-A",
        "language": "pl",
        "title": "ARTE Reportage",
        "subtitle": "USA: Dyskryminacja na porodówce",
```

Reply:
Ah, I know what happened. The titles *used to be* the opposite way, but they were changed while the PR was stalled.

Compare <https://web.archive.org/web/20201103170616/https://www.arte.tv/pl/videos/100103-000-A/usa-dyskryminacja-na-porodowce/>, where ‘ARTE Reportage’ is shown in smaller font, as the subordinate title. I’m pretty sure this agrees with what the API responded with back then.

I felt it was strange that I could somehow have swapped the two, since I distinctly remember copy-pasting the metadata straight from the output of the failing testcase. But that makes more sense.